### DISCUSSION TITLE
Suggestion closing marker (```) conflicts with Markdown code block closing marker

### DISCUSSION THREAD

**Discussion by scy** [community/community#76840] [Pull Requests]
### Select Topic Area

Bug

### Body

When code reviewing a Markdown document, you might come across a case where you’re adding a comment to multiple lines of code, and they include a code block, e.g.

````markdown
## How to touch grass

```sh
touch grass
```

It’s nice!
````

Now, suppose you’d like to use the “add suggestion” feature to propose a change. The comment field will now contain something like this:

````markdown
```suggestion
## How to touch grass

```sh
touch grass
```

It’s nice!
```
````

As you can see, there’s now two ` ``` ` closing markers in the comment, and sure enough, GitHub will think your suggestion ends after the _first_ one (i.e. below `touch grass`). `It’s nice` won’t be interpreted as part of the suggestion anymore.

Now, there’s a way to work around this. When a code block is supposed to contain a closing marker as part of its content, Markdown allows you to increase the number of backticks around that block to remove the ambiguity. For our suggestion, that could look like this:

`````markdown
````suggestion
## How to touch grass

```sh
touch grass
```

It’s nice!
````
`````

(Note that ` ````suggestion ` now starts with four backticks, and there’s also four backticks in the last line.)

However, **the “add suggestion” button should do this automatically**, and right now it doesn’t. I had to manually fix the Markdown created by the button.

I think most people won’t know about this Markdown syntax feature at all, and the suggestion button should do the right thing nevertheless. The algorithm could be something like this:

* When “add suggestion” is clicked, iterate over the selected lines.
* If a line matches ``^`{3,}$``, remember the number of backticks it has in a set or similar data structure.
* After having seen all the lines, find the smallest number ≥ 3 that’s not in the set.
* Use this as the number of backticks for ` ```suggestion ` and the closing marker.

**Comment by matevz** [comment]
Why not simply have the four-backticks variant there by default?

**Reply by scy** [reply]
This might solve 95 % of all cases, but still fails if the original Markdown contains four backticks as a separator. A universal solution as outlined above doesn't have that problem.

**Comment by nicwolff** [comment]
I'd have liked this today – I had to find this discussion to learn about four backticks! (And then I needed five backticks to quote four backticks in a code block explaining how to quote three backticks!)

**Comment by msakrejda** [comment]
I also had to find this discussion to learn about four backticks It's also documented [here](https://github.github.com/gfm/#fenced-code-blocks), but I stumbled on this discussion first. I agree the proposed behavior would be an improvement.
